src/main/java/com/yegor256/Jhome.java

```diff
@@ -85,7 +94,17 @@ public Path java() {
      *  {@code Optional<Path>} is also a solution here.
      */
     public Path javac() {
-        return this.path(String.format("bin/javac%s", Jhome.extension()));
+        final Path path = this.path(String.format("bin/javac%s", Jhome.extension()));
+        if (Files.exists(path)) {
+            return path;
```

Comment:
@volodya-lombrozo see this: https://www.yegor256.com/2015/01/21/if-then-throw-else.html